### PR TITLE
initial adopt of latest kapitan changes

### PR DESCRIPTION
**Pull request by Moep90**
adopting https://github.com/kapicorp/kapitan/pull/845

### CODE REVIEW

File: components/generators/argocd/__init__.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 def j2(filename, ctx):
     return render_jinja2_file(filename, ctx, search_paths=search_paths)
 
-inv = inventory()
+inv = Dict()
```
Comment (ramaro) [inline components/generators/argocd/__init__.py:17]: This should not be changed. `inv = inventory()` is still the right way.

File: components/generators/kubernetes/__init__.py [hunk 2]
```diff
@@ -632,7 +632,7 @@ def add_volume_mounts_from_configs(self):
         secrets = container.secrets.items()
         for object_name, spec in configs:
             if spec is None:
-                raise CompileError(
+                raise ABORT_EXCEPTION_TYPE(
```
Comment (ramaro) [inline components/generators/kubernetes/__init__.py:635]: You don't need to use `ABORT_EXCEPTION_TYPE`, this is already set to `CompileError` in the kadet input type

**Comment by ademariag** [comment]
Hello @Moep90 , thank you for this, I was just going to work on it.

Is this ready to be merged (apart from the ci error?)

**Comment by Moep90** [comment]
> Hello @Moep90 , thank you for this, I was just going to work on it.
> 
> Is this ready to be merged (apart from the ci error?)

Hey @ademariag.
There are some parts still missing from rendering like with 0.30 but overall it should work

**Comment by ademariag** [comment]
I'm ok to wait, let's finish it to that it renders the same

**Comment by ademariag** [comment]
Obsoleted by https://github.com/kapicorp/kapitan-reference/pull/133

**Pull request closed without merging**